src/nvim/indent_c.c

```diff
@@ -508,29 +508,21 @@ get_indent_nolabel (     /* XXX */
  *   label:	if (asdf && asdfasdf)
  *		^
  */
-static int skip_label(linenr_T lnum, char_u **pp)
+static int get_indent_after_label(linenr_T lnum, char_u **pp) FUNC_ATTR_NONNULL_ALL
 {
-  char_u      *l;
-  int amount;
-  pos_T cursor_save;
+  char_u *l = ml_get(lnum);
 
-  cursor_save = curwin->w_cursor;
-  curwin->w_cursor.lnum = lnum;
-  l = get_cursor_line_ptr();
-  /* XXX */
   if (cin_iscase(l, FALSE) || cin_isscopedecl(l) || cin_islabel()) {
-    amount = get_indent_nolabel(lnum);
-    l = after_label(get_cursor_line_ptr());
-    if (l == NULL)              /* just in case */
-      l = get_cursor_line_ptr();
-  } else {
-    amount = get_indent();
-    l = get_cursor_line_ptr();
+    // We use "ml_get(lnum)" rather than reusing "l" because each call to
+    // ml_get invalidates the results of previous calls (and "cin_islabel()"
+    // calls it).
+    char_u *i = after_label(ml_get(lnum));
+    *pp = (i == NULL) ? (char_u *)1 : i;
+    return get_indent_nolabel(lnum);
```

Comment:
`get_indent_nolabel(lnum)` also calls `ml_get(lnum)`, so at this point `*pp` is an invalid pointer.